CMakeLists.txt

```diff
@@ -157,11 +157,11 @@ endif( ${CMAKE_BUILD_TYPE} MATCHES "Release" )
 
 # Set compiler flags
 if(${CMAKE_COMPILER_IS_GNUCXX})
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -funroll-loops -Wall -std=c++11")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -funroll-loops -Wall -std=c++11 -Wl,--no-as-needed -ldl")
```

Comment:
what about clang and intel (@roigcarlo)?

in the linearsolversapp it is added for every compiler except MSVC
@oberbichler can you comment please?

Reply:
https://github.com/KratosMultiphysics/Kratos/blob/ebf673f520148a5e167c7bc4058c8f0b063c9a61/applications/LinearSolversApplication/CMakeLists.txt#L35-L37

Reply:
If I remember correctly we just use the Intel link advisor to get the correct settings. For me it would make sense to use the GCC settings also for the other non-MSVC compilers. At least as an initial guess.

Reply:
I added flag to Clang as well. Let's see if it compiles

Reply:
I am not sure if clang is gonna accept those flags out of the box, let me check

Reply:
@roigcarlo You are totally right. It failed to compile. These flags are GCC specific. I will remove them.

Reply:
> I am not sure if clang is gonna accept those flags out of the box, let me check

interestingly clang didn't reject them but could no longer find OpenMP :thinking: 
~~~
CMAKE_CXX_FLAGS = -Werror -Wno-deprecated-declarations -std=c++11 -O0 -Wall -fPIC -funroll-loops -Wall -Wno-unused-local-typedef -Wno-unknown-pragmas -Wl,--no-as-needed -ldl
CMAKE_C_FLAGS =  -fPIC -funroll-loops -Wall -Wno-unknown-pragmas -Wl,--no-as-needed -ldl
-- Found PythonInterp: /usr/bin/python3.8 (found suitable version "3.8.10", minimum required is "3.8") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.8.so
Python version used for the interface will be 3.8
-- Standard install dir /__w/Kratos/Kratos/bin/Custom
"KRATOS_SHARED_MEMORY_PARALLELIZATION" not defined, defaulting to "OpenMP"
-- Found OpenMP_C: -fopenmp=libomp (found version "4.5") 
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_CXX_FOUND) (found version "4.5")
CMake Error at CMakeLists.txt:327 (message):
  OpenMP could not be found!
~~~

not sure, clang seems to be sensitive wrt finding OpenMP

in any case, for me this is ready

Reply:
@philbucher @roigcarlo is it OK now to merge?

Reply:
yes ok for me, @roigcarlo can you please approve if it is ok for you?